### ISSUE TITLE
GTextField.setFocus() causes exception when highlighting empty textbox

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roloww/gui4processing#5]
```
What steps will reproduce the problem?
1.  Create a GTextField and put it inside its own GWindow.
2.  Programmatically set the GTextField to be in focus 
(GTextField.setFocus(true)).
3.  Before typing any text into the GTextField, click & drag anywhere inside 
the 
     GWindow NOT inside of the GTextField (nor inside of any other GComponent on the GWindow).

What is the expected output? What do you see instead?
The initial String inside the GTextField should highlight.  Instead, an 
Exception occurs:

java.lang.StringIndexOutOfBoundsException: String index out of range: -1
    at java.lang.String.substring(String.java:1937)
    at guicomponents.GTextField.cursorPosition(GTextField.java:268)
    at guicomponents.GTextField.draw(GTextField.java:772)
    at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:848)
    at processing.core.PApplet$RegisteredMethods.handle(PApplet.java:841)
    at processing.core.PApplet.handleDraw(PApplet.java:1645)
    at processing.core.PApplet.run(PApplet.java:1530)

What version of the product are you using? On what operating system?
V1.7.2 on Mac OS X Snow Leopard 10.6.3. 

Please provide any additional information below.
Single-clicking inside of the GTextField before attempting to highlight 
prevents the Exception from occurring.  After setFocus() has been called, 
typing characters into the GTextField works fine, yet
highlighting with the mouse by clicking outside of the GTextField causes the 
Exception.

```

Original issue reported on code.google.com by `michael....@gmail.com` on 30 Oct 2011 at 3:12

**Comment by GoogleCodeExporter** [comment]
```
This problem occurs wherever the GTextField object is placed. This bug will be 
fixed in version 1.7.3 which I will release ASAP. Thanksfor reporting the 
problem.
```

Original comment by `p.k.la...@gmail.com` on 2 Nov 2011 at 8:42
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Now fixed with release V1.7.3
```

Original comment by `p.k.la...@gmail.com` on 2 Nov 2011 at 6:15
- Changed state: **Verified**

**Comment by GoogleCodeExporter** [comment]
```
Wow!  Thanks for your promptness in addressing this issue, Peter.  It works
great now.
```

Original comment by `michael....@gmail.com` on 3 Nov 2011 at 3:56
